eZ/Publish/Core/Persistence/Cache/AbstractInMemoryHandler.php

```diff
@@ -99,6 +98,9 @@ final protected function getCacheValue(
         $this->logger->logCacheMiss($arguments ?: [$id], 3);
 
         $object = $backendLoader($id);
+
+        $this->cacheIndicesValidator->validate($keyPrefix, $object, $cacheIndexes);
```

Comment:
Please ensure this logic is executed only if kernel is in debug mode

Reply:
Then may I suggest we only create this `CacheIndicesValidator` in debug mode, and treat it as an optional dependency?

Reply:
@adamwojs @Steveb-p `CacheIndicesValidator` created only in debug mode as of now: https://github.com/ezsystems/ezplatform-kernel/pull/367/commits/18dcebca9725baa4452b1eac7a25b9d8d51b2773